Sources/NIOFileSystem/FileSystemProtocol.swift

```diff
@@ -696,6 +1002,32 @@ extension FileSystemProtocol {
     ///   - options: Options used to create the directory.
     ///   - execute: A closure which provides access to the directory and its path.
     /// - Returns: The result of `execute`.
+    public func withTemporaryDirectory<Result>(
+        prefix: NIOFilePath? = nil,
+        options: OpenOptions.Directory = OpenOptions.Directory(),
+        execute: (_ directory: DirectoryFileHandle, _ path: NIOFilePath) async throws -> Result
+    ) async throws -> Result {
+        try await self.withTemporaryDirectory(prefix: prefix.map { FilePath($0) }, options: options) {
+            directory,
+            path in
```

Comment:
`swift-format` makes bad choices sometimes:

```suggestion
        try await self.withTemporaryDirectory(
            prefix: prefix.map { FilePath($0) }, 
            options: options
        ) { directory, path in
```